src/pages/workspaces/workspace/analysis/modals/ExportAnalysisModal/ExportAnalysisModal.state.ts

```diff
@@ -0,0 +1,119 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { useWorkspaces } from 'src/components/workspace-utils'
+import {
+  AnalysisProviderDependencies,
+  analysisProviderDependencies
+} from 'src/dependencies/analysis/analysis-provider-resolver'
+import { Composable } from 'src/dependencies/dependency-core'
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData'
+import { useMetricsEvent } from 'src/libs/ajax/metrics/useMetrics'
+import Events, { extractCrossWorkspaceDetails } from 'src/libs/events'
+import { useCancellation } from 'src/libs/react-utils'
+import LoadedState from 'src/libs/type-utils/LoadedState'
+import { WorkspaceInfo, WorkspaceWrapper } from 'src/libs/workspace-utils'
+import {
+  AnalysisFile,
+  getDisplayName
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { ToolLabel } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+
+
+export interface AnalysisExportStateDeps {
+  analysisProviders: Pick<AnalysisProviderDependencies, 'AnalysisProvider'>
+}
+
+export type LoadedAnalysisNames = LoadedState<string[], unknown>
+
+export interface AnalysisExportStateArgs {
+  sourceWorkspace: WorkspaceWrapper
+  printName: string
+  toolLabel: ToolLabel
+}
+
+export interface AnalysisExportState {
+  workspaces: WorkspaceWrapper[]
+  selectWorkspace: (workspaceId: string) => void
+  selectedWorkspace: WorkspaceInfo | null
+  existingAnalysisNames: LoadedAnalysisNames
+  copyAnalysis: (newName: string) => void
+  pendingCopy: LoadedState<true, unknown>
+}
+
+export type AnalysesExportStateHook = (args: AnalysisExportStateArgs) => AnalysisExportState
+
+export type ComposableAnalysisExportStateHook = Composable<
+    AnalysisExportStateDeps,
+    AnalysesExportStateHook
+    >
+
+export const errors = {
+  badWorkspace: 'Selected Workspace does not exist',
+  noWorkspace: 'No workspace selected'
+}
+
+export const useAnalysisExportState: ComposableAnalysisExportStateHook = {
+  compose: (deps: AnalysisExportStateDeps) => {
+    const useHook: AnalysesExportStateHook = (hookArgs): AnalysisExportState => {
+      const { sourceWorkspace, printName, toolLabel } = hookArgs
+      const { AnalysisProvider } = deps.analysisProviders
+      const { captureEvent } = useMetricsEvent()
```

Comment:
the rest of the hook code here can be skimmed past to line 114.  It is from a pending IA PR and not central to the Composable DI patterns being demontrated.  It does however show a couple of examples of using LoadedData and useLoadedData helper hook for those who are curious.